msteams-platform/toolkit/debug-your-Teams-app-test-tool.md

```diff
@@ -0,0 +1,478 @@
+    BOT_ID=
+    BOT_PASSWORD=
+    TEAMSFX_NOTIFICATION_STORE_FILENAME=.notification.testtoolstore.json
+    ```
+
+1. Go to **EXPLORER** > **env**.
+1. Create a **.env.testtool** file and add the following code:
+
+    ```json
+    // .env.testtool
+    # This file includes environment variables that can be committed to git. It's gitignored by default because it represents your local development environment
+    # Built-in environment variables
+    TEAMSFX_ENV=testtool
+    # Environment variables used by test tool
+    TEAMSAPPTESTER_PORT=56150
+    ```
+
+1. If you have custom environment variables, set their values in **.env.testtool** or **.env.testtool.user**.
+
+1. Add either an OpenAI key or Azure OpenAI key and endpoint in **.env.testtool.user**.
+
+    ```json
+    # SECRET_OPENAI_API_KEY=***********
+    SECRET_AZURE_OPENAI_API_KEY=***********
+    SECRET_AZURE_OPENAI_ENDPOINT=<https://your-openai-service-name.openai.azure.com/>
+    ```
+
+1. Go to **package.json** and add the following code under the `scripts` property:
+
+    ```json
+    "scripts": {
+        ... 
+        "dev:teamsfx:testtool": "env-cmd --silent -f .localConfigs.testTool npm run dev", 
+        "dev:teamsfx:launch-testtool": "env-cmd --silent -f env/.env.testtool teamsapptester start", 
+        ... 
+    },
+    ```
+
+1. From the left pane, select **Run and Debug** (`Ctrl+Shift+D`) and select **Debug in Test Tool** in dropdown list.
+
+   :::image type="content" source="../assets/images/teams-toolkit-v2/debug/select-debug-in-test-tool.png" alt-text="Screenshot shows the option to select debug in test tool.":::
+
+Test Tool successfully debugs your existing bot.
+
+## FAQ
+
+<details>
+<summary>How can I test my bot if the Test Tool doesn't support its features?</summary>
+
+You can always use the [Teams client](https://teams.microsoft.com/) to test the features that Test Tool doesn't support. Select the option **Debug (Edge)** or **Debug (Chrome)** to test your application in the Teams client.
+<br>
+&nbsp;
+</details>
+<details>
+<summary>How would I know if Test Tool doesn't support features in my bot?</summary>
+
+The Test Tool shows a warning message in conversation and log panel when it detects unsupported features.
+
+:::image type="content" source="../assets/images/teams-toolkit-v2/debug/features-not-supported.png" alt-text="Screenshot shows the warning message of an unsupported feature.":::
+<br>
+&nbsp;
+</details>
+<details>
+<summary>Does Microsoft recommend using only the Test Tool for testing bot applications?</summary>
+
+No. We always recommend users to test their bot application in the Teams client before moving the application to production environment.
+<br>
+&nbsp;
+</details>
+
+## Code sample
+
+|Sample name | Description | Node.js |
+|----------------|-----------------|--------------|
+| Test Tool | A sample bot app to explore Test Tool. | [View](https://github.com/OfficeDev/TeamsFx-Samples/tree/v3/test-tool-sample-app) |
```

Comment:
The sample name should be Test Tool Sample App?

Reply:
Done. Changed.